react/MegaMenu/components/Submenu.tsx

```diff
@@ -222,6 +225,66 @@ const Submenu: FC<ItemProps> = observer((props) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [departmentActive, collapsibleStates]
   )
+
+  const itemsWithNoSubCategories = useMemo(
```

Comment:
Looking at the file, it seems that there's a fair bit of duplication between `itemsWithSubCategories` / `items` – We might want to extract the following, somehow: https://github.com/syatt-io/mega-menu/blob/dc7e3cda2f89c9862008510c68c1e17d10f13d55/react/MegaMenu/components/Submenu.tsx#L119-L136